### ISSUE TITLE
Main loop stops looping in some error conditions

### ISSUE THREAD

**Issue by sebbacon** [opensafely/job-runner#15]
I just saw this, and had to restart the container in order to get the loop running again:

```
job-runner_1  | 2020-07-09 13:02:55,064 INFO        job#45     Reported error 3 (usage: entrypoint.py [-h] script
job-runner_1  | entrypoint.py: error: unrecognized arguments: /tmp/outputs2/seb-please-delete-me-test-research-master-dummy/input.csv
job-runner_1  |  id job#45) to job server
job-runner_1  | 2020-07-09 13:02:55,065 ERROR       job#45     usage: entrypoint.py [-h] script
job-runner_1  | entrypoint.py: error: unrecognized arguments: /tmp/outputs2/seb-please-delete-me-test-research-master-dummy/input.csv
job-runner_1  | Traceback (most recent call last):
job-runner_1  |   File "/app/runner/__init__.py", line 350, in report_result
job-runner_1  |     job = future.result(
job-runner_1  |   File "/root/.pyenv/versions/3.8.3/lib/python3.8/concurrent/futures/_base.py", line 432, in result
job-runner_1  |     return self.__get_result()
job-runner_1  |   File "/root/.pyenv/versions/3.8.3/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
job-runner_1  |     raise self._exception
job-runner_1  | runner.exceptions.ScriptError: usage: entrypoint.py [-h] script
job-runner_1  | entrypoint.py: error: unrecognized arguments: /tmp/outputs2/seb-please-delete-me-test-research-master-dummy/input.csv
```

**Comment by sebbacon** [comment]
Thinking about it, I'm going to guess this is a race condition. [Locks are available in the pebble library](https://pythonhosted.org/Pebble/).

**Comment by sebbacon** [comment]
The main loop actually quite after an assertion error "The only currently-supported mechanism..."

**Comment by sebbacon** [comment]
Closed because #40 is v similar
